api/envoy/extensions/filters/network/http_connection_manager/v3/http_connection_manager.proto

```diff
@@ -802,6 +802,10 @@ message HttpConnectionManager {
   // [#not-implemented-hide:]
   // [#extension-category: envoy.http.header_validators]
   config.core.v3.TypedExtensionConfig typed_header_validation_config = 50;
+
+  // Append the `x-forwarded-port` header with the port value client used to connect to Envoy. It
+  // will be ignored if the `x-forwarded-port` header has been set by any proxy in front of Envoy.
+  bool append_x_forwarded_port = 51;
```

Comment:
@adisuissa should this be a ServerHeaderTransformation so the user can specify overwrite vs append if absent vs pass through?  if not I think it'd be good for the naming to include an if_absent

Reply:
That's a great question.
It probably shouldn't be [ServerHeaderTransformation](https://github.com/envoyproxy/envoy/blob/e368c3e5e6dcbcb893364dda99b71ff9d68f3158/api/envoy/extensions/filters/network/http_connection_manager/v3/http_connection_manager.proto#L66), as the default is to overwrite, but I can see how it can be a similar enum with a passthrough default-value.

Unfortunately I wasn't able to find a spec for this header. When I initially looked at this field, the closest thing I found in Envoy was [append_x_forwarded_host](https://github.com/envoyproxy/envoy/blob/8a9a7d28f7b468c984cf1da5e09edbb423bce603/api/envoy/config/route/v3/route_components.proto#L1180), which is also defined as a `bool`.

I can see use-cases of not-appending, and appending, but I'm not sure if there's a use-case for overriding it. If there is one, or if you think we should be prudent here, then I'd say that using an enum makes sense.